spring-integration-smb/src/main/java/org/springframework/integration/smb/filters/LastModifiedSmbFileListFilter.java

```diff
@@ -0,0 +1,139 @@
+ * {@link SmbFile#getLastModified()} is less than the {@link #age} in comparison
+ * with the current time.
+ * <p>
+ *     The resolution is done in seconds.
+ * </p>
+ * When {@link #discardCallback} is provided, it called for all the rejected files.
+ *
+ * @author Adama Sorho
+ *
+ * @since 6.2
+ */
+public class LastModifiedSmbFileListFilter implements DiscardAwareFileListFilter<SmbFile> {
+
+	private static final long ONE_SECOND = 1000;
+	private static final long DEFAULT_AGE = 60;
+	private volatile long age = DEFAULT_AGE;
+
+	@Nullable
+	private Consumer<SmbFile> discardCallback;
+
+	public LastModifiedSmbFileListFilter() {
+	}
+
+	/**
+	 * Construct a {@link LastModifiedSmbFileListFilter} instance with provided {@link #age}.
+	 * Defaults to 60 seconds.
+	 * @param age the age in seconds.
+	 */
+	public LastModifiedSmbFileListFilter(long age) {
+		this.age = age;
+	}
+
+	/**
+	 * Set the age that the files have to be before being passed by this filter.
+	 * If {@link SmbFile#getLastModified()} plus age is greater than the current time, the file
+	 * is filtered. The resolution is seconds.
+	 * Defaults to 60 seconds.
+	 * @param age the age in seconds.
+	 * @param unit the timeUnit.
+	 */
+	public void setAge(long age, TimeUnit unit) {
+		this.age = unit.toSeconds(age);
+	}
+
+	/**
+	 * Set the age that the files have to be before being passed by this filter.
+	 * If {@link SmbFile#getLastModified()} plus age is greater than the current time, the file
+	 * is filtered. The resolution is seconds.
+	 * Defaults to 60 seconds.
+	 * @param age the age in seconds.
+	 */
+	public void setAge(Duration age) {
+		setAge(age.getSeconds());
+	}
+
+	/**
+	 * Set the age that the files have to be before being passed by this filter.
+	 * If {@link SmbFile#getLastModified()} plus age is greater than the current time, the file
+	 * is filtered. The resolution is seconds.
+	 * Defaults to 60 seconds.
+	 * @param age the age in seconds.
+	 */
+	public void setAge(long age) {
+		setAge(age, TimeUnit.SECONDS);
+	}
+
+	@Override
+	public void addDiscardCallback(@Nullable Consumer<SmbFile> discardCallback) {
+		this.discardCallback = discardCallback;
+	}
+
+	@Override
+	public List<SmbFile> filterFiles(SmbFile[] files) {
+		List<SmbFile> list = new ArrayList<>();
+		long now = System.currentTimeMillis() / ONE_SECOND;
+		for (SmbFile file: files) {
+			if (fileIsAged(file, now)) {
+				list.add(file);
+			}
+			else if (this.discardCallback != null) {
+				this.discardCallback.accept(file);
+			}
+		}
+
+		return list;
+	}
+
+	@Override
+	public boolean accept(SmbFile file) {
+		if (fileIsAged(file, System.currentTimeMillis() / ONE_SECOND)) {
+			return true;
+		}
+		else if (this.discardCallback != null) {
+			this.discardCallback.accept(file);
+		}
+
+		return false;
+	}
+
+	private boolean fileIsAged(SmbFile file, long now) {
+		return file.getLastModified() / ONE_SECOND + this.age <= now;
```

Comment:
This SMB API does not provide an `Instant` representation, so, we have to stick with millis.
However those blank lines around members an removal of `volatile` are still valid concerns.